### PR TITLE
Handle Event - Coin Detail

### PR DESCRIPTION
**Pull request by TranTinh030892**
## Related Tickets
- [#Task 79252: Handle Event - Coin Detail](https://edu-redmine.sun-asterisk.vn/issues/79252)
## WHAT
- Handle Event - Coin Detail
## Review Checklist

Category | View Point | Description | Expected Reviewer Answer | Self review | Reviewer2 (name)
--- | --- | --- | --- | --- | ---
Conventions | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/blob/master/eng/android/coding_convention_android_kotlin.md | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>
Redmine | Does the ticket follow Sun* Redmine working process?  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md| YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>
Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |  | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>
## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/example/fina/screen/detail/ChartExt.kt [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+fun FragmentDetailBinding.setupChartAppearance() {
+    val xAxis = this.lineChart.xAxis
+    xAxis.setDrawGridLines(false)
+    xAxis.setDrawAxisLine(false)
+    xAxis.setDrawLabels(false)
+
+    val leftAxis = this.lineChart.axisLeft
+    leftAxis.setDrawGridLines(true)
+    leftAxis.setDrawAxisLine(false)
+    leftAxis.gridColor = Color.parseColor("#FFDBDDE8")
```
Comment (xuanthu2002) [inline app/src/main/java/com/example/fina/screen/detail/ChartExt.kt]: Hard code.

**Review by MaxHolmes221B — APPROVED** [review]

**Review by xuanthu2002 — APPROVED** [review]

File: app/build.gradle
Comment (ghost) [inline app/build.gradle:1]: Bỏ change file này

File: app/src/main/java/com/example/fina/utils/ext/ViewExt.kt [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+fun ProgressBar.showWithDelay(
+    lineChart: LineChart,
+    delayMillis: Long = 700L,
+    action: () -> Unit,
+) {
+    lineChart.clear()
+    lineChart.setNoDataText("")
+    this.visibility = View.VISIBLE
+    CoroutineScope(Dispatchers.Main).launch {
+        delay(delayMillis)
```
Comment (ghost) [inline app/src/main/java/com/example/fina/utils/ext/ViewExt.kt]: Project 1 không sử dụng Coroutine em nhé

**Review by ghost — CHANGES_REQUESTED** [review]
.

File: app/src/main/java/com/example/fina/screen/detail/DetailPresenter.kt [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+            dataSet.setDrawFilled(true)
+            dataSet.fillColor = ContextCompat.getColor(context, R.color.light_blue)
+            dataSet.setDrawCircles(false)
+            dataSet.setDrawValues(false)
+
+            return LineData(dataSet)
+        }
+
+        override fun onStart() {
+            TODO()
```
Comment (ghost) [inline app/src/main/java/com/example/fina/screen/detail/DetailPresenter.kt]:
```suggestion
            /* no-op */
```

File: app/src/main/java/com/example/fina/screen/detail/DetailPresenter.kt [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+        }
+
+        override fun onStop() {
+            TODO()
```
Comment (ghost) [inline app/src/main/java/com/example/fina/screen/detail/DetailPresenter.kt]:
```suggestion
            
```

File: app/src/main/java/com/example/fina/screen/detail/DetailPresenter.kt [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+import com.example.fina.data.model.PriceRecord
+import com.example.fina.data.repository.CoinRepository
+import com.example.fina.data.repository.OnResultListener
+import com.example.fina.utils.ExtraParams
+import com.example.fina.utils.MarketCapInterval
+import com.github.mikephil.charting.data.Entry
+import com.github.mikephil.charting.data.LineData
+import com.github.mikephil.charting.data.LineDataSet
+
+class DetailPresenter internal constructor(private val coinRepository: CoinRepository?) :
```
Comment (ghost) [inline app/src/main/java/com/example/fina/screen/detail/DetailPresenter.kt]: Repository không được phép null em
```suggestion
class DetailPresenter internal constructor(private val coinRepository: CoinRepository) :
```

File: app/src/main/java/com/example/fina/screen/detail/ChartExt.kt [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+        val price = e.y
+        val sdf = SimpleDateFormat("dd/MM/yyyy HH:mm:ss", Locale.getDefault())
+        val date = Date(timestamp * 1000) // Convert from seconds to milliseconds
+        val formattedDate = sdf.format(date)
+
+        priceTextView.text = "$price"
+        dateTextView.text = "$formattedDate"
+
+        val transformer = lineChart.getTransformer(lineChart.data.getDataSetByIndex(0).axisDependency)
+        val pixelPos = FloatArray(2)
```
Comment (ghost) [inline app/src/main/java/com/example/fina/screen/detail/ChartExt.kt]: Code rối quá, format lại tí e

**Review by ghost — CHANGES_REQUESTED** [review]

**Review by ghost — APPROVED** [review]

**Pull request merged**